File: 创建型模式/单例模式/php实现/Singleton.php
<?php 
class Singleton{
    private static $instance;
    private function __construct()
    {
        // 防止被实例化
    }
    
    public static function getInstance(){
        // 使用getInstance  确保只有一个实例被生成
        if (self::$instance == null){
            self::$instance = new Singleton();
        }
        return self::$instance;
    }
}
?>

File: 创建型模式/单例模式/java实现/Singleton.java
// java实现单例模式 懒汉式 线程安全 但是会过多占用内存

public class Singleton {
    private static Singleton instance;
    // 让构造函数为private这样该类就不会被实例化
    private Singleton(){};
    public static synchronized Singleton getInstance(){
        if (instance == null){
            instance = new Singleton();
        }
        return instance;
    }
}